### ISSUE TITLE
"Mark as read" option on messages essentially deletes messages

### ISSUE THREAD

**Issue by sacrevert** [BiologicalRecordsCentre/iRecord#1484]
See some discussion here:
https://github.com/BiologicalRecordsCentre/NPMS/issues/290 

Given that in email systems "mark as read" does just that, and does not delete or hide messages, I think it would be preferable have some sort of archive option rather than a confusing "mark as read" option. An archived messages list would then allow users to view and filter old messages if required.

**Comment by kitenetter** [comment]
@sacrevert a potential difficulty here is that the comments often do not make sense except in the context of the records that they apply to. And also a lot of 'comments' are relatively trivial automated messages that people are very unlikely to need to go back to.

I agree that the "mark as read" button is perhaps misleading, and as things stand it could usefully be relabelled as "dismiss comment".

But I wonder if a better solution would be to find a way of enabling users to filter their records for "all records that have had a comment added that is not an automated comment". And we might also want to split out "comments for queried records" from "comments for verification decisions" and from "comments by other users".

I'm not sure how complex it would be to provide such filters, but I don't think much would be gained by being able to go to the comments themselves in isolation from the records.

**Comment by sacrevert** [comment]
Naive question, but why would a comment ever be unlinked from the record if related to (which you seem to imply)? Wouldnt the comment always provide a link back to the record? Agree that filtering on content would be useful however.

**Comment by kitenetter** [comment]
@sacrevert I think I was misunderstanding what you were proposing. Were you suggesting that we simply add a "view previously read notifications" option to the existing list of notifications, so that you could go back to any older notification and then click the "View record" button to see it in context? If so that seems achievable, although the lists could end up being very long.

**Comment by sacrevert** [comment]
Yes, that's essentially it. I agree that the list might be long, but if it was searchable that wouldn't really matter, and an actual "delete" option (which could work on a filtered grid basis) could also be supplied

**Comment by johnvanbreda** [comment]
One thing to bear in mind when considering this issue is that the notifications table is enormous, so having some sort of auto-delete of old notifications would be preferable from that perspective (e.g. > 1 year). The comments associated with the notification would still persist.

In order to support @kitenetter's proposed comment filters, we would need to update the Elasticsearch index with a count of comments for the various types.

**Comment by Sam-Amy** [comment]
I think this has also just come up in a forum post: "I removed a notification without thinking and now I cannot figure out what record it referred to. Is there any way of tracking the record down other than going back through all my records?". Is there any way to do this?

**Comment by kitenetter** [comment]
@Sam-Amy do they know what type of notification it was? If it was for a query added to a record then they go to the "My records" view and filter for queried records. If it was an auto-response for a record cleaner rule then that is more difficult, but it seems less likely that they would be bothered about re-finding an auto response?

**Comment by sacrevert** [comment]
@johnvanbreda Thanks John. If this were to go ahead, then some sort of auto-deletion for auto-responses would be sensible. Actual human queries could presumably be retained though, as I imagine there are far fewer of these. (I appreciate these can be found as per @kitenetter suggestion on filtering, but it seems logical to me to keep these messages as a notifications archive too)
